drivers/backup/portworx/portworx.go

```diff
@@ -0,0 +1,477 @@
+	}
+	return nil
+}
+
+func (d *pxbackup) CreateCloudCredential(orgID string, name string) error {
+	cc := d.getCloudCredentialManager()
+	req := &api.CloudCredentialCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  name,
+			OrgId: orgID,
+		},
+		CloudCredential: &api.CloudCredentialInfo{},
+	}
+
+	req.CloudCredential.Type = api.CloudCredentialInfo_AWS
+	req.CloudCredential.Config = &api.CloudCredentialInfo_AwsConfig{
+		AwsConfig: &api.AWSConfig{
+			AccessKey: awsAccessKey,
+			SecretKey: awsSecretKey,
+		},
+	}
+
+	_, err := cc.Create(context.Background(), req)
+	if err != nil {
+		fmt.Printf("Unable to create credential %v\n", err)
+	}
+	return err
+}
+
+func (d *pxbackup) InspectCloudCredential(orgID string, name string) (*api.CloudCredentialInspectResponse, error) {
+	cc := d.getCloudCredentialManager()
+	resp, err := cc.Inspect(
+		context.Background(),
+		&api.CloudCredentialInspectRequest{Name: name, OrgId: orgID},
+	)
+	return resp, err
+}
+
+func (d *pxbackup) EnumerateCloudCredential(orgID string) (*api.CloudCredentialEnumerateResponse, error) {
+	cc := d.getCloudCredentialManager()
+	resp, err := cc.Enumerate(
+		context.Background(),
+		&api.CloudCredentialEnumerateRequest{OrgId: orgID},
+	)
+
+	return resp, err
+}
+
+func (d *pxbackup) DeleteCloudCredential(orgID string, name string) error {
+	cc := d.getCloudCredentialManager()
+	req := &api.CloudCredentialDeleteRequest{
+		Name:  name,
+		OrgId: orgID,
+	}
+
+	_, err := cc.Delete(context.Background(), req)
+	if err != nil {
+		fmt.Printf("Unable to delete credential %v\n", err)
+	}
+	return nil
+}
+
+func (d *pxbackup) CreateCluster(name string, orgID string, labels map[string]string, cloudCredential string, pxToken string, config string) (*api.ClusterCreateResponse, error) {
+	cluster := d.getClusterManager()
+	req := &api.ClusterCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:   name,
+			OrgId:  orgID,
+			Labels: labels,
+		},
+		Cluster: &api.ClusterInfo{
+			PxConfig: &api.PXConfig{
+				AccessToken: pxToken,
+			},
+			Kubeconfig:      config,
+			CloudCredential: cloudCredential,
+		},
+	}
+	resp, err := cluster.Create(context.Background(), req)
+	if err != nil {
+		fmt.Printf("Unable to create cluster %v\n", err)
+	}
+	return resp, err
+}
+
+func (d *pxbackup) InspectCluster(orgID string, name string) (*api.ClusterInspectResponse, error) {
+	cluster := d.getClusterManager()
+	resp, err := cluster.Inspect(
+		context.Background(),
+		&api.ClusterInspectRequest{OrgId: orgID, Name: name},
+	)
+	return resp, err
+}
+
+func (d *pxbackup) EnumerateCluster(orgID string) (*api.ClusterEnumerateResponse, error) {
+	cluster := d.getClusterManager()
+	resp, err := cluster.Enumerate(
+		context.Background(),
+		&api.ClusterEnumerateRequest{OrgId: orgID},
+	)
+	return resp, err
+}
+
+func (d *pxbackup) DeleteCluster(orgID string, name string) (*api.ClusterDeleteResponse, error) {
+	cluster := d.getClusterManager()
+	resp, err := cluster.Delete(
```

Comment:
```suggestion
	return cluster.Delete(
```
Same for other places